include/xtensor/xfft.hpp

```diff
@@ -0,0 +1,294 @@
+#ifndef XFFT_HPP
+#define XFFT_HPP
+
+#include "xmath.hpp"
+#include "xcomplex.hpp"
+#include "xutils.hpp"
+#include "xaxis_slice_iterator.hpp"
+#include "xio.hpp"
+#include "xview.hpp"
+
+namespace xt 
+{
+    namespace fft
+    {
+        template<typename T = double, typename E1, typename E2>
+        xt::xarray<std::complex<T>> fftconvolve(E1&& xvec, E2&& yvec, std::ptrdiff_t axis = -1);
+        
+        namespace detail
+        {
+            template<typename T>
+            T PI()
+            {
+                constexpr double PI = 3.141592653589793238463;
+                return PI;
+            }
+
+            size_t reverseBits(size_t x, int n)
+            {
+                size_t result = 0;
+                for (int i = 0; i < n; i++, x >>= 1)
+                    result = (result << 1) | (x & 1U);
+                return result;
+            }
+
+            bool IsPower2(size_t vecSize) noexcept
+            {
+                return ((vecSize & (vecSize - 1)) == 0);
+            }
+
+            template<typename T, typename E1>
+            auto transformRadix2(const E1& _data)
+            {
+
+                xt::xarray<std::complex<T>> data = _data;
+
+                // Length variables
+                const size_t n(data.shape(0));
+                int levels = std::floor(std::log2(n));
+
+                // Trignometric table
+                auto i = xt::linspace<T>(0, n / 2 - 1, n / 2) * std::complex<T>(0, 1);
+                //I would like to do this but there appears to be a compiler error in CPP14
+                //Works in CPP17 and MSVC CPP14
+                //Linker error when using numeric_constants in this header
+                //auto expTable = xt::eval(xt::exp(-2 * i * ::xt::numeric_constants<T>().PI / n));
+                auto expTable = xt::eval(xt::exp(-2 * i * PI<T>() / n));
+
+                // Bit-reversed addressing permutation
+                for (size_t i = 0; i < n; i++)
+                {
+                    const size_t j = reverseBits(i, levels);
+                    if (j > i)
+                    {
+                        std::swap(data(i), data(j));
+                    }
+                }
+
+                // Cooley-Tukey decimation-in-time radix-2 FFT
+                // Could probably be reimplemented using stride views...
+                for (size_t size = 2; size <= n; size *= 2)
+                {
+                    const size_t halfsize = size / 2;
```

Comment:
To my annoyance we use `std::size_t` instead of `size_t`

Reply:
Haha. I was going to make a joke about that but in fear of it getting lost in translation I will say... Ok... I'll keep that in mind.